### DISCUSSION TITLE
Clock: step through half cycles

### DISCUSSION THREAD

**Discussion by me-nkr** [hneemann/Digital#1253] [Q&A] [answered]
Say I have an edge-triggered component that gets triggered on the falling edge. If I want to test and verify that it doesn't get triggered on the rising edge, I have to be able to stop the clock in half cycle and assert the state right? But I didn't see the option to do this in Digital's test functionality. Am I missing something or is this a feature that can be and will be added?

Unrelated ( not opening another discussion since it seems trivial )
- Suggestion: option to stop test script from running since accidental infinite while loops can freeze the app.

**Comment by me-nkr** [comment, marked as answer]
It seems like I asked too early.

While `C` completes a full cycle `1` and `0` can be used to tick-tock ( half steps ) the clock.

It would be nice if someone could confirm this though : )

**Reply by hneemann** [reply]
You are right! :smile:
